sympy/solvers/ode.py

```diff
@@ -4003,6 +4002,31 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+
+def _check_substitution_type(eq, func):
+    # Check if some derivate of positive order can be substituted as g(x) = f^(order_to_subs) (x)
+    x = func.args[0]
+    f = func.func
+    order_to_subs = 0
+    for nsubs_try in range(1,ode_order(eq)):
```

Comment:
Suggest `D = Dummy()` before loop and then reuse in loop.

Reply:
Did it.